lib/mongodb.js

```diff
@@ -1508,7 +1531,8 @@ MongoDB.prototype.count = function count(modelName, where, options, callback) {
  */
 MongoDB.prototype.replaceById = function replace(modelName, id, data, options, cb) {
   if (this.debug) debug('replace', modelName, id, data);
-  const oid = this.coerceId(modelName, id, options);
+  // SPIKE: every id should be converted to ObjectID before query
+  const oid = ObjectID(id);
```

Comment:
See my comments above. IMO, the conversion to ObjectID should be applied only if the id property is defined with data type ObjectID.

It would be great if we could re-use existing bits from the infrastructure for coercing arbitrary model properties.